validation/validator.py

```diff
@@ -69,14 +82,40 @@ def check_categorical(col_name):
 for num_col in numeric_cols:
     check_numeric(num_col)
 
-
-categorical_cols = list(set(list(df.drop(['Notes', 'refID', 'StudyName'], axis=1).columns)) - set(numeric_cols))
+excluded_cols = ['Notes', 'refID', 'StudyName']
+categorical_cols = list(set(list(df.drop(excluded_cols, axis=1).columns)) - set(numeric_cols))
 
 for cat_col in categorical_cols:
     check_categorical(cat_col)
 
 
-# Passed successfully
+# Data validation passed successfully
 
 os.rename(opts.data_file_path, os.path.join(opts.output_file_dir,
                                             'InterventionOutcomesRFull.csv'))
+
+
+# Process codebook for updating
+def filter_codebook_cols(codebook_df):
+    '''
+    Find the rows in the codebook_df which contain the columns and descriptions of interest.
+
+    :param: codebook_df:downloaded raw codebook as a pandas dataframe
+    :returns: codebook_validated:list
+    '''
+    codebook_validated = []
+    for col in codebook_df.iloc[:, 0]:
+        if (col in categorical_vals.keys()) or (col in numeric_cols) or (col in excluded_cols):
+            codebook_validated.append(col)
+
+    return codebook_validated
+
+
+# Only keep column names that were filtered
+codebook_df = codebook_df[codebook_df.iloc[:, 0].isin(
```

Comment:
this line is a little difficult to parse, I won't hold it up for this pull request, but may be confusing for future wanderers.